app/src/main/java/com/google/android/sambadocumentsprovider/auth/AuthActivity.java

```diff
@@ -0,0 +1,169 @@
+    @Override
+    public void onClick(View view) {
+      tryAuth();
+    }
+  };
+
+  private final OnTaskFinishedCallback<Void> callback = new OnTaskFinishedCallback<Void>() {
+    @Override
+    public void onTaskFinished(@Status int status, @Nullable Void item, @Nullable Exception e) {
+      progressDialog.dismiss();
+
+      if (status == SUCCEEDED) {
+        setResult(RESULT_OK);
+        finish();
+      } else {
+        Log.i(TAG, "Authentication failed: ", e);
+
+        final Context context = getApplicationContext();
+        if (e instanceof AuthFailedException) {
+          Toast.makeText(
+                  context,
+                  getApplicationContext().getString(R.string.credential_error),
+                  Toast.LENGTH_SHORT).show();
+        } else {
+          Toast.makeText(
+                  context,
+                  getApplicationContext().getString(R.string.credential_error),
+                  Toast.LENGTH_SHORT).show();
+        }
+      }
+    }
+  };
+
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    final Context context = getApplicationContext();
+    mShareManager = SambaProviderApplication.getServerManager(context);
+    mClient = SambaProviderApplication.getSambaClient(context);
+
+    Intent authIntent = getIntent();
+    String shareUri = authIntent.getStringExtra(SHARE_URI_KEY);
+
+    prepareUI(shareUri);
+  }
+
+  private void prepareUI(String shareUri) {
+    mSharePathEditText = (EditText) findViewById(R.id.share_path);
+    mUsernameEditText = (EditText) findViewById(R.id.username);
+    mDomainEditText = (EditText) findViewById(R.id.domain);
+    mPasswordEditText = (EditText) findViewById(R.id.password);
+
+    CheckBox mPasswodCheckbox = (CheckBox) findViewById(R.id.needs_password);
+    mPinShareCheckbox = (CheckBox) findViewById(R.id.pin_share);
+
+    mSharePathEditText.setText(shareUri);
+    mSharePathEditText.setEnabled(false);
+
+    mPasswodCheckbox.setVisibility(View.GONE);
+    mPinShareCheckbox.setVisibility(View.VISIBLE);
+
+    Button mLoginButton = (Button) findViewById(R.id.mount);
+    mLoginButton.setText(getResources().getString(R.string.login));
+    mLoginButton.setOnClickListener(mLoginListener);
+
+    final Button cancel = (Button) findViewById(R.id.cancel);
+    cancel.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        finish();
+      }
+    });
+  }
+
+  private void tryAuth() {
+    progressDialog = ProgressDialog.show(
+            this, null, getResources().getString(R.string.authenticating), true);
+
+    if (TextUtils.isEmpty(mUsernameEditText.getText())
+            || TextUtils.isEmpty(mPasswordEditText.getText())) {
+      Toast.makeText(this, "Username and password can't be empty!", Toast.LENGTH_SHORT).show();
```

Comment:
Don't hard code user visible strings inline. Use resources.

Also you can extract username & password into local variables to avoid calling `getText()` multiple times.